### ISSUE TITLE
Caching using the CachingFrameAdapter makes the app exit

### ISSUE THREAD

**Issue by igorkulman** [Caliburn-Micro/Caliburn.Micro#285]
I started developing a Windows Phone 8.1 XAML app, using Caliburn.Micro 3beta2. 

When I do not use the CachingFrameAdapter (using the standard `container.RegisterNavigationService(rootFrame);`) the app works fine.  But I want to cache the VMs, so I switched to `container.RegisterNavigationService(rootFrame, cacheViewModels: true);` which creates a strange problem.

Here is the complete source code (just 3 VMs + 3 Views): https://dl.dropboxusercontent.com/u/73642/IIHF.zip

I start the app, it opens a screen with pivot. I wait for the data to load, select an item in the first pivot pivot and the app navigates to another screen. I press back, the app returns to the previous screen. 

Now when I start to swipe the pivot the app either
- crashes with `Unspecified error`
- just closes with `The program '[2324] IIHF.exe' has exited with code -1 (0xffffffff).`
- just closes with `The program '[2704] IIHF.exe' has exited with code -1073741819 (0xc0000005) 'Access violation'.`

Here is a video showing this

![cm](https://cloud.githubusercontent.com/assets/4746070/12393430/d75fa83a-bdf5-11e5-8b55-ecd768eb17d4.gif)

This occurs every 1 of 3 times I start the app. I think the VM caching must be the problem, because everything works without it enabled and it also works when I cache the View using (`NavigationCache=Enabled`).

**Comment by nigel-sampson** [comment]
When you're using the `CachingFrameAdapter` do you still have `NavigationCache=Enabled` on the views?

**Comment by igorkulman** [comment]
No, not caching the Views, just having the ViewModels cached.

**Comment by nigel-sampson** [comment]
Still looking into this, I have noticed that navigating to a blank `TestViewModel` rather than `EventSelectorDetailViewModel` causes the crash rate to drop to a level I had almost thought had solved the problem. Can you see if that's the case for you as well?

**Comment by nigel-sampson** [comment]
Also removing `EventSelectorViewModel,SelectedCategory` seems to resolve it. Still looking into why.

**Comment by igorkulman** [comment]
I also tried to remove the Pivot binding to `EventSelectorViewModel.SelectedCategory` and I still get crashes sometimes. 

And I discovered another problem. When I do not use the `CachingFrameAdapter` just `NavigationCache=Required` I sometimes, but not very often get this strange exception:

![screenshot 35](https://cloud.githubusercontent.com/assets/4746070/12613641/4acd88e4-c4fb-11e5-8378-bb02a197e2f2.png)

Looks like the TwoWay binding to Pivot seems to be quite a problem for some reason.
